merlin/core/dispatch.py

```diff
@@ -489,16 +496,21 @@ def concat(objs, **kwargs):
 
 def make_df(_like_df=None, device=None):
     """Return a DataFrame with the same dtype as `_like_df`"""
-    if not cudf or isinstance(_like_df, (pd.DataFrame, pd.Series)):
-        return pd.DataFrame(_like_df)
-    elif isinstance(_like_df, (cudf.DataFrame, cudf.Series)):
+    if not cudf or device == "cpu" or isinstance(_like_df, (pd.DataFrame, pd.Series)):
+        # move to pandas need it on CPU (host memory)
+        # can be a cudf, cupy or numpy Series
+        if cudf and isinstance(_like_df, (cudf.DataFrame, cudf.Series)):
+            # move to cpu
+            return _like_df.to_pandas()
+        if cp and isinstance(_like_df, cp.ndarray):
+            return pd.DataFrame(cp.asarray(_like_df))
```

Comment:
`cupy.asarray` returns a cupy array which might not work well with the pandas DataFrame contructor. An additional test for what is expected would help clarify what we'd like to convert from/to.

Reply:
Actually that particular is the only way you can avoid the `TypeError: Implicit conversion to a NumPy array is not allowed. Please use .get() to construct a NumPy array explicitly.` When translating from cupy to pandas. Good call on the test though, I will put in a test to illustrate that.